lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -50,7 +50,12 @@ def #{tag_type}_list=(new_tags)
               parsed_new_list = ActsAsTaggableOn.default_parser.new(new_tags).parse
 
               if self.class.preserve_tag_order? || parsed_new_list.sort != #{tag_type}_list.sort
-                set_attribute_was('#{tag_type}_list', #{tag_type}_list)
+                if Rails.version < "6.0.0.beta1"
```

Comment:
`if Rails::MAJOR::VERSION < 6`

`Rails.version` is just a string.

```
"6.0.0" < "6.0.0.beta1"
=> true
```

which is not what you want